chapter07/chapter07.adoc

```diff
@@ -0,0 +1,276 @@
+
+==== Types of Metric Registries
+
+MicroProfile Metrics defines several types of registries, categorized by their scope:
+
+- Application Scope (`MetricRegistry.Type.APPLICATION`): Contains custom metrics that are specific to the application. These are typically the metrics that developers explicitly create and register to monitor application-specific behaviors.
+
+- Base Scope (`MetricRegistry.Type.BASE`): Contains metrics that are fundamental and common across all MicroProfile applications. These metrics provide basic information about the underlying JVM and application server.
+
+- Vendor Scope (`MetricRegistry.Type.VENDOR`): Contains metrics that are specific to the implementation of the MicroProfile platform being used. These metrics offer insights into vendor-specific features and optimizations.
+
+==== Instrumenting Microservices with MicroProfile Metrics
+
+Instrumenting microservices with MicroProfile Metrics enables developers to gain detailed insights into their application's operational health and performance. This level of observability is essential for maintaining scalable and resilient microservice architectures in dynamic environments.
+Tracking response time using @Timed
+
+MicroProfile Metrics also allows you to track a method's response time as a timed metric. The code example below shows how to use the @Timed annotation to track the response time.
+
+[source]
+----
+import org.eclipse.microprofile.metrics.annotation.Timed;
+// …
+
+public class ProductResource {
+
+    // …
+    // Expose the response time as a timer metric
+    @Timed(name = "productLookupTime",
+            tags = {"method=getProduct"},
+            absolute = true, 
+            description = "Time needed to lookup for a products")
+    public Product getProduct(@PathParam("id") Long productId) {
+        return productService.getProduct(productId);
+    }
+
+    // …
+----
+  
+It will expose a metric called `productLookupTime`, which will track the amount of time spent in the `getProduct()` method in milliseconds.
+You can visit the following URL `++https://localhost:<port>/metrics?scope=application++` (Replace `<port>` with the actual port where the server is running) to see the Response time of this method as below:
+
+[source]
+----
+…
+# HELP productLookupTime_seconds_max Time needed to lookup for a products
+# TYPE productLookupTime_seconds_max gauge
+productLookupTime_seconds_max{method="getProduct",mp_scope="application",} 0.002270643
+…
+----
+
+==== Tracking number of invocations
+MicroProfile Metrics also allows you to track the number of invocations of a method as a counter metric. The code example below shows how to use the `@Counted` annotation to track the invocation count.
+
+[source, java]
+----
+import org.eclipse.microprofile.metrics.Metrics;
+import org.eclipse.microprofile.metrics.MetricType;
```

Comment:
won't need this line (MetricType) for the example

Reply:
Updated.